Cds 48450 tag infra and gitops clusters (#1600)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/advanced/multiserv-multienv.md

You can deploy services to infrastructures in an environment or environment group using filtered list.

Make sure that you have added tags when creating your environment in the **Configuration** tab.

![tag environments](./static/tag-environments.png)

:::note

Currently, this feature is behind the feature flag, `CDS_FILTER_INFRA_CLUSTERS_ON_TAGS`. Make sure these related feature flags, `MULTI_SERVICE_INFRA`, `NG_SVC_ENV_REDESIGN`, `ENV_GROUP`, `OPTIMIZED_GIT_FETCH_FILES` are enabled. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

1. In your CD stage, select **Service**.
2. In **Select Services**, select the services you want to deploy.
3. Select **Continue**.
4. In the **Environments** tab, enable **Deploy to multiple Environments or Infrastructures**.
5. Select **Environment** or **Environment Group**.  
   You can select one or more environments or environment groups, and then one or more infrastructures in each environment or environment group.
6. In **Infrastructures**, select **Deploy to filtered list**, and then select **+ Add Filters**.
7. In **FILTERS ON ENTITIES**, select **Infrastructure**.
8. In **Type**, select: 

>    * **All** to deploy the selected service(s) to all infrastructures within the environment or environment group.

Author: ```suggestion
   * **All** to deploy the selected services to all infrastructures within the environment or environment group.
```

   * **Tags**, and then enter the tags in **CONDITION** to deploy the selected service(s) to the infrastructures with these tags. 
        

>      Select **Any** to deploy to infrastructures that has any of these tags. 

Author: ```suggestion
     Select **Any** to deploy the selected service(s) to infrastructures that has any of these tags. 
```

>      Select **Any** to deploy the selected service(s) to infrastructures that has any of these tags. 

Author: ```suggestion
     Select **Any** to deploy the selected services to infrastructures that have any of these tags. 
```

>      Select **All** to deploy the selected service(s) to infrastructures having all these tags. 

Author: ```suggestion
     Select **All** to deploy the selected services to infrastructures having all these tags. 
```

You can deploy services to the GitOps clusters in an environment using filtered list.

Make sure that you have added tags when creating your environment in the **Configuration** tab.

:::note

Currently, this feature is behind the feature flag, `CDS_FILTER_INFRA_CLUSTERS_ON_TAGS`. Make sure these related feature flags, `MULTI_SERVICE_INFRA`, `NG_SVC_ENV_REDESIGN`, `ENV_GROUP`, `OPTIMIZED_GIT_FETCH_FILES` are enabled. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

1. In your pipeline, create a **Deploy** stage with the **GitOps** option enabled.
2. In your CD stage, select **Service**.
3. In **Select Services**, select the services you want to deploy.
4. Select **Continue**.
5. In the **Environments** tab, enable **Deploy to multiple Environments or Clusters**.
6. Select **Environment** or **Environment Group**.  
   You can select one or more environments or environment groups, and then one or more infrastructures in each environment or environment group.
7. In **Clusters**, select **Deploy to filtered list**, and then select **+ Add Filters**.
8. In **FILTERS ON ENTITIES**, select **Clusters**.
9. In **Type**, select: 

>    * **All** to deploy the selected service(s) to all clusters within the environment or environment group.

Author: ```suggestion
   * **All** to deploy the selected services to all clusters within the environment or environment group.
```

>    * **Tags**, and then enter the tags in **CONDITION** to deploy the selected service(s) to the clusters with these tags. 

Reviewer: **Tags**, and then enter the tags in **CONDITION** to deploy the selected services to the clusters with these tags. 
Global comment - Remove the parenthesis, see https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/s-es

>    * **Tags**, and then enter the tags in **CONDITION** to deploy the selected service(s) to the clusters with these tags. 

Author: ```suggestion
   * **Tags**, and then enter the tags in **CONDITION** to deploy the selected services to the clusters with these tags. 
```

>      Select **Any** to deploy to clusters that has any of these tags. 

Author: ```suggestion
     Select **Any** to deploy the selected service(s) to clusters that has any of these tags. 
```

>      Select **Any** to deploy the selected service(s) to clusters that has any of these tags. 

Reviewer: ```suggestion
     Select **Any** to deploy the selected service(s) to clusters that have any of these tags. 
```

>      Select **Any** to deploy the selected service(s) to clusters that has any of these tags. 

Author: ```suggestion
     Select **Any** to deploy the selected services to clusters that have any of these tags. 
```

>      Select **All** to deploy the selected service(s) to clusters having all these tags. 

Author: ```suggestion
     Select **All** to deploy the selected services to clusters having all these tags. 
```
